Source/WebKit/UIProcess/WebProcessCache.cpp

```diff
@@ -59,8 +63,38 @@ WebProcessCache::WebProcessCache(WebProcessPool& processPool)
     platformInitialize();
 }
 
-bool WebProcessCache::canCacheProcess(WebProcessProxy& process) const
+bool WebProcessCache::canCacheProcess(WebProcessProxy& process, CheckSuspendedState checkSuspendedState) const
 {
+    WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: memory usage: %lu", process.processID(), process.memoryUsage());
+
+#if PLATFORM(IOS)
+    if (WTF::ramUnused() < 256 * MB) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because there is not enough free memory, free RAM: %lu", process.processID(), WTF::ramUnused());
+        return false;
+    }
+
+    if (process.hasMediaActivity()) {
```

Comment:
A Media activity is just a process assertion. A process wouldn't be able to enter the process cache if it still had media playing in it. Therefore, this check doesn't make much sense.